### PR TITLE
feat(claim_feed): create new endpoint to list using a cursor and remove useless table joins

### PR DESCRIPTION
**Pull request by omareloui**
## Description

Please provide a brief summary of the changes in this PR.

## Type of Change

- [ ] Bug fix
- [ ] New feature
- [ ] Test case
- [ ] Documentation update
- [ ] Other (please describe)

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works (optional)
- [ ] I have added necessary documentation (if applicable)

## 4- steps to test ?

Please provide a brief summary of the steps required to test the changes in this PR.

## 5- results ?

Please provide a brief summary of the results after testing the changes in this PR.

## 7- screenshots ?

Please provide screenshots of the results after testing the changes in this PR.

### CODE REVIEW

File: src/dao/api.dao.ts [hunk 1]
```diff
@@ -335,6 +345,73 @@ export class NodeDao {
     }
   };
 
+  async getFeedEntriesV3(limit: number, cursor: string | null, query: string | null) {
```
Comment (gvelez17) [inline src/dao/api.dao.ts:348]: ok good so we can compare with previous one, did you do any tests?

File: src/dao/api.dao.ts [hunk 2]
```diff
@@ -335,6 +345,73 @@ export class NodeDao {
...
+          WHERE
+            n."entType" != 'CLAIM'
+            AND e.label != 'source'
+            AND c."effectiveDate" IS NOT NULL
+            AND c.statement IS NOT NULL
+            AND n.name IS NOT NULL
+            AND n.name != ''
+            AND (
+              c.subject ILIKE COALESCE(${query}, '%') OR
+              c.statement ILIKE COALESCE(${query}, '%') OR
```
Comment (gvelez17) [inline src/dao/api.dao.ts:376]: this is going to be slow, we might want to try without anything that does not have an index

Comment (gvelez17) [reply src/dao/api.dao.ts:376]: it is ok for the query i think but, we could also query first for name or url and fallback to statement only if we don't have those kind of matches.  anyway its ok, but i think our other problems are really separate from this

* initial feed is slow (could precache, does not require search, why is it slow if its just most recent?)

* when going to graph we should NOT do query but search on id exactly only

anyway this is ok, its fine for a try, but just mentioning those are probably more pressing problems

File: src/dao/api.dao.ts (hunk 1 above)
Comment (omareloui) [reply src/dao/api.dao.ts:348]: Yes, all tested and it's working as expected

File: src/dao/api.dao.ts (hunk 2 above)
Comment (omareloui) [reply src/dao/api.dao.ts:376]: When I tested it, the diff wasn't that big. We can improve it more by text indexing the fields we need to search with, and as you said. rely on fields more suitable for searching. I've removed some fields that need to join a lot of tables just to search in them without really showing any data that is used to search. That was the huge gain came from, removing unused joins. Now the request takes less than 1s, ofc that's not the best but at least it's a good step.

After using this endpoint we'll have a better idea on what is the bottleneck. The we can improve with the new information

Comment (gvelez17) [reply src/dao/api.dao.ts:430]: ok!  ha i just edited same code, sorry about that!  but yes lets get it in that's fine, if this is working i can approve, don't wait for me in any case

**Review by gvelez17 — APPROVED** [review]

**Pull request merged**